src/fides/api/alembic/migrations/versions/548a1ac26771_fideslang_3.0_remove_references.py

```diff
@@ -0,0 +1,244 @@
+"""remove data_qualifier
+
+Revision ID: 548a1ac26771
+Revises: 848a8f4125cf
+Create Date: 2023-12-08 22:07:38.617038
+
+"""
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = "548a1ac26771"
+down_revision = "848a8f4125cf"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    ## Remove data qualifier references
+    op.drop_index("ix_ctl_data_qualifiers_fides_key", table_name="ctl_data_qualifiers")
+    op.drop_index("ix_ctl_data_qualifiers_id", table_name="ctl_data_qualifiers")
+    op.drop_table("ctl_data_qualifiers")
+    op.drop_column("ctl_datasets", "data_qualifier")
+    op.drop_index("ix_ctl_systems_name", table_name="ctl_systems")
+    op.drop_column("privacydeclaration", "data_qualifier")
+    op.drop_constraint("purpose_constraint", "tcf_purpose_overrides", type_="unique")
+
+    ## Remove registry references
+    op.drop_index("ix_ctl_registries_fides_key", table_name="ctl_registries")
+    op.drop_index("ix_ctl_registries_id", table_name="ctl_registries")
+    op.drop_table("ctl_registries")
+    op.drop_column("ctl_systems", "registry_id")
+
+    ## Remove data use fields
+    op.drop_column("ctl_data_uses", "legal_basis")
+    op.drop_column("ctl_data_uses", "special_category")
+    op.drop_column("ctl_data_uses", "legitimate_interest")
+    op.drop_column("ctl_data_uses", "legitimate_interest_impact_assessment")
+    op.drop_column("ctl_data_uses", "recipients")
+
+    ## Remove dataset fields
+    op.drop_column("ctl_datasets", "retention")
+    op.drop_column("ctl_datasets", "joint_controller")
+    op.drop_column("ctl_datasets", "third_country_transfers")
+
+    ## Remove system fields
+    op.drop_column("ctl_systems", "joint_controller")
+    op.drop_column("ctl_systems", "data_responsibility_title")
+    op.drop_column("ctl_systems", "third_country_transfers")
+    op.drop_column("ctl_systems", "data_protection_impact_assessment")
+
+
+def downgrade():
+    ## Add back in data qualifier references
+    op.create_unique_constraint(
+        "purpose_constraint", "tcf_purpose_overrides", ["purpose"]
+    )
+    op.add_column(
+        "privacydeclaration",
+        sa.Column("data_qualifier", sa.VARCHAR(), autoincrement=False, nullable=True),
+    )
+    op.create_index("ix_ctl_systems_name", "ctl_systems", ["name"], unique=False)
+    op.add_column(
+        "ctl_datasets",
+        sa.Column("data_qualifier", sa.VARCHAR(), autoincrement=False, nullable=True),
+    )
+    op.create_table(
+        "ctl_data_qualifiers",
+        sa.Column("id", sa.VARCHAR(length=255), autoincrement=False, nullable=False),
```

Comment:
in the downrevs, I'd update all these to be nullable=True so it can actually be downgraded

Reply:
thank you, great catch!